### PR TITLE
User profile store improvements

### PR DESCRIPTION
**Pull request by nathancoleman**
- Initial unit tests in UserProfileStoreTest.java
- Delete functionality in UserProfileStore.java
- Updated README to include instructions for easily installing Maven with Homebrew

### CODE REVIEW

File: src/data/proxy/UserProfileStore.java [hunk 1]
```diff
@@ -21,23 +21,30 @@ public UserProfileStore() {
...
     public void write(UserProfile profile) {
         userProfiles.put(profile.getId(), profile);
     }
+
+    /**
+     * Remove a user profile from storage.
+     *
+     * @param id The String id for the UserProfile to delete
+     */
+    public void delete(String id) { userProfiles.remove(id); }
```
Comment (sethau) [inline src/data/proxy/UserProfileStore.java]: Good idea for delete() - I don't know why I didn't think to add that in the first place!

Let's be sure to adhere to a consistent coding standard. Can you please match the format used elsewhere?

File: tst/data/proxy/UserProfileStoreTest.java [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
-    // TODO
+
+    /**
+     * Tests null return for missing UserProfile id.
+     */
+    @Test
+    public void testGetProfileNonExistent() {
+        TestUserProfileStore userProfileStore = new TestUserProfileStore();
+        Map<String, UserProfile> userProfiles = new HashMap<String, UserProfile>();
+        userProfileStore.setUserProfiles(userProfiles);
```
Comment (sethau) [inline tst/data/proxy/UserProfileStoreTest.java]: I don't think this is really necessary here. If you're trying to test that a null is returned when "get" is called with a non-existent ID, then you can just create a UserProfileStore instance and immediately call the get method. Furthermore, this is the only case where you use the TestUserProfileStore class (and by proxy, the protected variable "userProfiles"), so I don't think that this is worth expanding the visibility of the private userProfiles attribute of UserProfileStore.

I would advise resetting that attribute's visibility to private and removing the TestUserPRofileStore class here for the sake of better encapsulation and simpler code.

P.S. - Even if you did leave it as protected (not saying you should), you don't have to have a separate test class just for that. Even UserProfileStoreTest could access protected variables.

File: src/data/proxy/UserProfileStore.java [hunk 3]
```diff
@@ -9,7 +9,7 @@
  * UserProfileStore manages access to the stored user profiles.
  */
 public class UserProfileStore {
-    private Map<String, UserProfile> userProfiles;
+    protected Map<String, UserProfile> userProfiles;
```
Comment (sethau) [inline src/data/proxy/UserProfileStore.java]: See comment below (in UserProfileStoreTest) regarding visibility.

File: src/data/proxy/UserProfileStore.java (hunk 1 above)
Comment (nathancoleman) [inline src/data/proxy/UserProfileStore.java]: Ugh. Super OCD about this stuff - turned off the editor setting that fucked up my precious code.

File: tst/data/proxy/UserProfileStoreTest.java (hunk 2 above)
Comment (nathancoleman) [inline tst/data/proxy/UserProfileStoreTest.java]: I wasn't aware you could access private vars in a JUnit test class :+1:  - addressed.

Comment (sethau) [inline tst/data/proxy/UserProfileStoreTest.java]: You can't access private variables, but you can access protected or package-visible variables. My point was that you didn't need a SECOND class in this test file just to access that variable IFF you had left it as protected. Does that make sense?

**Comment by sethau** [comment]
Looks good, Ship It!

**Pull request merged**